clap_derive/src/utils/ty.rs

```diff
@@ -26,11 +28,15 @@ impl Ty {
             t(Bool)
         } else if is_generic_ty(ty, "Vec") {
             t(Vec)
+        } else if cfg!(feature = "unstable-array") && is_array_ty(ty) {
+            t(Array)
```

Comment:
Forgot another use case: how do people opt-out?

I don't think we are really documenting it at the moment but the way to opt-out of the other built-in types is by fully qualifying the path but I'm unsure if there is a way to do this.

There might be times when a user wants to parse a single argument into one of these types.  We already provide an example for a tuple (`examples/typed-derive.rs`) and I recently stepped someone through how to do it with `Vec`, its not a stretch to assume people will need to do that for this case as well.

Maybe the opt-out is `std::option::Option<[...]>` and they manually mark it as `required = true`?  Whatever we decide, we should at least have a test for it.